pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -1395,6 +1395,14 @@ func (c *PodAffinityChecker) satisfiesPodsAffinityAntiAffinity(pod *v1.Pod, node
 
 // PodToleratesNodeTaints checks if a pod tolerations can tolerate the node taints
 func PodToleratesNodeTaints(pod *v1.Pod, meta algorithm.PredicateMetadata, nodeInfo *schedulercache.NodeInfo) (bool, []algorithm.PredicateFailureReason, error) {
+	if nodeInfo == nil || nodeInfo.Node() == nil {
+		return false, []algorithm.PredicateFailureReason{ErrNodeUnknownCondition}, nil
+	}
+
+	if nodeInfo.Node().Spec.Unschedulable {
```

Comment:
maybe use `node := nodeInfo.Node()`

Reply:
seems duplicated, nodeInfo.Node() is used only once :).

Reply:
it is used to check if it is nil, but it is just a nit, we can leave it as is.